### DISCUSSION TITLE
What amplitude is saved in spikes.amplitudes?

### DISCUSSION THREAD

**Discussion by yang-yiling** [petersenpeter/CellExplorer#99] [Q&A] [answered]
Hi, I would like to ask, inside the "spikes" structure, what amplitude values are saved in the field "spikes.amplitudes"? I understood that averaged peak-to-peak amplitudes are saved in the field "spikes.peakVoltage", one value per cell, and that "spikes.amplitudes" seem to store some amplitude values per cell and per detected spike. However, I cannot relate "spikes.peakVoltage" and "spikes.amplitudes". Thank you for any information.

**Comment by petersenpeter** [comment, marked as answer]
Hi

spikes.amplitudes are imported from Kilosort. I believe it is a measure coming from the templates. There is one amplitude per spike.

spikes.peakVoltage is extracted from the actual waveforms from the raw data and is in units of µV. The peak voltage is calculated from the average waveform.

**Reply by yang-yiling** [reply]
Hi Dr. Petersen, thank you for your timely reply. Is it possible to extract the peak-to-peak voltage/amplitude (in units of uV) from each detected spike? As I understood it, this information is not saved by CellExplorer and one would have to go back to the raw data? Thank you for any tips.

**Reply by petersenpeter** [reply]
Hi

There is a dedicated function (getWaveformsFromDat) for this. From the session folder:
```
session = loadSession
spikes = loadSpikes
spikes = getWaveformsFromDat(spikes,session,'nPull',Inf,'keepWaveforms_filt',true,'keepWaveforms_raw',true)
```
The parameter nPull defines how many waveforms to extract.

**Reply by yang-yiling** [reply]
Thank you very much for the information!
